ztp/siteconfig-generator/siteConfig/siteConfigBuilder.go

```diff
@@ -478,3 +485,84 @@ func (scbuilder *SiteConfigBuilder) splitYamls(yamls []byte) ([][]byte, error) {
 
 	return resources, nil
 }
+
+func filterExtraManifests(dataMap map[string]interface{}, filter *Filter) (map[string]interface{}, error) {
+	// return if there's no filter initialized
+	if filter == nil {
+		return dataMap, nil
+	}
+
+	// check if there is an intersection in filenames
+	if filter.Include != nil && filter.Exclude != nil && len(filter.Exclude) > 0 && len(filter.Include) > 0 {
```

Comment:
A simpler sanity check would be:
- if the default is "include", the include list must be empty (because it is only valid to exclude specific files in this mode)
- elif the default is "exclude", the exclude list must be empty (because it is only valid to include specific files in this mode)

(I think that would cover everything, without traversing the lists?)

Reply:
Yep I think so too. But the purpose of the traversal is get the names of all the offending files to help users debug quickly -- `Can not have same file names in both exclude and include lists. Files %s are present in both include and exclude lists", stringFiles`

Reply:
If we short-circuit it to say "When in exclude mode you can't have anything in the exclude list", we don't even need to report the list contents though...

Reply:
done